src/DynamoCoreWpf/ViewModels/Watch3D/AttachedProperties.cs

```diff
@@ -63,5 +64,62 @@ public static bool GetHasTransparencyProperty(UIElement element)
         {
             return (bool) element.GetValue(HasTransparencyProperty);
         }
+
+        /// <summary>
+        /// A flag indicating whether the geometry is frozen
+        /// </summary>
+        public static readonly DependencyProperty IsFrozenProperty = DependencyProperty.RegisterAttached(
+            "IsFrozen",
+            typeof(bool),
+            typeof(GeometryModel3D),
+            new PropertyMetadata(false, IsFrozenPropertyChanged));
+
+        public static void SetIsFrozen(UIElement element, bool value)
+        {
+            element.SetValue(IsFrozenProperty, value);
+        }
+
+        public static bool GetIsFrozen(UIElement element)
+        {
+            return (bool)element.GetValue(IsFrozenProperty);
+        }
+
+
+        /// <summary>
+        /// This updates the transparency on the Geometry object.
+        /// </summary>
+        /// <param name="obj">The object.</param>
+        /// <param name="e">The <see cref="DependencyPropertyChangedEventArgs"/> instance containing the event data.</param>
+        private static void IsFrozenPropertyChanged(DependencyObject obj, DependencyPropertyChangedEventArgs e)
+        {
+            if (obj is GeometryModel3D && obj.GetType() != typeof(BillboardTextModel3D))
+            {
+                var geom = (GeometryModel3D)obj;
+                var colors = geom.Geometry.Colors.ToArray();
+
+                for (int i = 0; i < colors.Length; i++)
+                {
+                    colors[i].Alpha = colors[i].Alpha * transparentGeometry;
+                }
+
+                geom.Geometry.Colors.Clear();
+                geom.Geometry.Colors.AddRange(colors);
+
+                var vertex = geom as DynamoGeometryModel3D;
```

Comment:
this variable name doesn't seem to make sense to me, can you rename it or add a comment explaining why it should be called vertex?